test_plans/khr/free_function_commands.asciidoc

```diff
@@ -0,0 +1,413 @@
+          DestT* dest);
+
+template <typename SrcT, int SrcDims, access_mode SrcMode, typename DestT>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          std::shared_ptr<DestT> dest);
+
+}
+```
+
+==== copy (accessors, device to device)
+
+```C++
+namespace sycl::khr {
+
+template <typename SrcT, int SrcDims, access_mode SrcMode,
+          typename DestT, int DestDims, access_mode DestMode>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          sycl::accessor<DestT, DestDims, DestMode, target::device> dest);
+
+template <typename SrcT, int SrcDims, access_mode SrcMode,
+          typename DestT, int DestDims, access_mode DestMode>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          sycl::accessor<DestT, DestDims, DestMode, target::device> dest);
+
+}
+```
+
+==== memset
+
+```C++
+namespace sycl::khr {
+
+void memset(sycl::handler& h, void* ptr, int value, size_t numBytes);
+
+void memset(sycl::queue q, void* ptr, int value, size_t numBytes);
+
+}
+```
+
+==== fill
+```C++
+namespace sycl::khr {
+
+template <typename T>
+void fill(sycl::handler& h, T* ptr, const T& pattern, size_t count);
+
+template <typename T, int Dims, access_mode Mode>
+void fill(sycl::handler& h,
+          sycl::accessor<T, Dims, Mode, target::device> dest,
+          const T& src);
+
+template <typename T>
+void fill(sycl::queue q, T* ptr, const T& pattern, size_t count);
+
+template <typename T, int Dims, access_mode Mode>
+void fill(sycl::queue q,
+          sycl::accessor<T, Dims, Mode, target::device> dest,
+          const T& src);
+
+}
+```
+
+For the `update_host`, `prefetch` and `mem_advise` operations, assert that they can be called without throwing an exception.
+
+==== update_host
+
+```C++
+namespace sycl::khr {
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::handler& h, accessor<T, Dims, Mode, target::device> acc);
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::queue q, accessor<T, Dims, Mode, target::device> acc);
+
+}
+```
+
+==== prefetch
+
+```C++
+namespace sycl::khr {
+
+void prefetch(sycl::handler& h, void* ptr, size_t numBytes);
+
+void prefetch(sycl::queue q, void* ptr, size_t numBytes);
+
+}
+```
+
+==== mem_advise
+
+```C++
+namespace sycl::khr {
+
+void mem_advise(sycl::handler& h, void* ptr, size_t numBytes, int advice);
+
+void mem_advise(sycl::queue q, void* ptr, size_t numBytes, int advice);
+
+}
+```
+
+=== Command and event barriers
+
+These tests should use `#ifdef SYCL_EXT_ONEAPI_ENQUEUE_BARRIER` so they can be skipped
+if feature is not supported. For each barrier function, enqueue some commands before and after enqueuing the barrier. Assert that the commands enqueued after the barrier do not execute until those enqueued before the barrier have completed. The barrier functions are the following:
```

Comment:
The KHR doesn't have a dependency on the oneAPI extension.  The `command_barrier` and `event_barrier` functions are expected to be supported by any device that supports the KHR.

```suggestion
For each barrier function, enqueue some commands before and after enqueuing the barrier. Assert that the commands enqueued after the barrier do not execute until those enqueued before the barrier have completed. The barrier functions are the following:
```